### ISSUE TITLE
Fix test__linter__get_violations_filter_rules test for when new rules are added

### ISSUE THREAD

**Issue by pwildenhain** [sqlfluff/sqlfluff#620]
> Are we ok with this test potentially failing again as we add more rules? I like that we test the functionality of passing "None" to `get_violations` but seems a little annoying to have to update this seemingly unrelated test as new rules come along
> 
> If only the person who had added this test had thought about this....(me)

Imo - no! 😂

There are a few cases where we have to do that whenever we add new rules anywhere though, so I don't have strong feelings. Either way, if we decide to do that can we do that in a separate PR to unblock master?

_Originally posted by @NiallRees in https://github.com/sqlfluff/sqlfluff/issues/617#issuecomment-739256792_

**Comment by alanmcruickshank** [comment]
I'm ok with some of the broader tests needing to be updated as we bring in more features. In some ways I think it's a *good* thing rather than a problem. It's a case of refining what we see as best practice.

**Comment by alanmcruickshank** [comment]
I'm going to close this issue because it hasn't come up again recently and I think we're getting to a more stable place with our rules estate.
